lib/eslint.js

```diff
@@ -657,10 +666,16 @@ module.exports = (function() {
         // process initial config to make it safe to extend
         config = prepareConfig(config || {});
 
-        ast = parse(text.replace(/^#!([^\r\n]+)/, function(match, captured) {
-            shebang = captured;
-            return "//" + captured;
-        }), config);
+        if (textOrParseResult.ast) {
+            // if a pregenerated ast was supplied, use it
+            ast = textOrParseResult.ast;
```

Comment:
I think perhaps the existing naming of `ast` here was misleading.  If I'm understanding correctly, the variable `ast` is really the result of the parsing (by default from espree), which includes tokens and some other things.  By assigning `ast` to `testOrParseResult.ast` you are losing the tokens `textOrParseResult.tokens`.  

So I think this should simply be 
```
ast = textOrParseResult;
```